### DISCUSSION TITLE
Is it possible to remove keycode import in videojs8 ?

### DISCUSSION THREAD

**Discussion by joanliky** [videojs/video.js#8729] [Q&A] [answered]
For security request, my project has been scanned by Sonatype Nexus Lifecycle. Which comes out with two issues by keycode:
<img width="357" alt="image" src="https://github.com/videojs/video.js/assets/85118665/9e490b3c-b857-4997-ba3a-256470574d23">

So, may i ask is it possible to remove keycode in videojs8?
if yes, may i ask how can i remove then?

**Comment by mister-ben** [comment]
Why would that package be a security issue? Your screen shot doesn't indicate that.

**Reply by mister-ben** [reply]
I saw an email notification of a reply that's apparently since been deleted that showed that it is mocha that's been flagged, which is a devDependency of keycode. That's a red herring in so far as a devDependency of a dependency of video.js won't be installed or executed by using video.js.

That said, we could probably remove keycode. That was useful when browsers were more inconsistent, but `KeyboardEvent.key` is supported by all browsers we support, so we could switch to that and remove the dependency. It would mean a refactor to use `event.key.toLowerCase() === 'x'` in place of `keycode.isEventKey(event, 'x')` etc.

**Reply by joanliky** [reply]
Thank so much for your reply.

so May i ask is it possible to remove that keycode at this stage? Or i should be waiting for next update?

**Reply by mister-ben** [reply, marked as answer]
https://github.com/videojs/video.js/pull/8735

Keycode has been removed in the next release.

**Reply by joanliky** [reply]
@mister-ben Thanks so much!

**Reply by joanliky** [reply]
> #8735
> 
> Keycode has been removed in the next release.

Hello, May i ask when this version will update~?

**Reply by gkatsev** [reply]
v8.14.1 and newer.

**Reply by joanliky** [reply]
Sorry for the confusion.
I mean may i ask when will the release ;')?

**Reply by gkatsev** [reply]
that version and a few newer ones are already available.

**Reply by gkatsev** [reply]
8.16.1 is now latest and includes this change

**Reply by joanliky** [reply]
Thanks a lot!

**Comment by joanliky** [comment]
> I saw an email notification of a reply that's apparently since been deleted that showed that it is mocha that's been flagged, which is a devDependency of keycode. That's a red herring in so far as a devDependency of a dependency of video.js won't be installed or executed by using video.js.
> 
> That said, we could probably remove keycode. That was useful when browsers were more inconsistent, but `KeyboardEvent.key` is supported by all browsers we support, so we could switch to that and remove the dependency. It would mean a refactor to use `event.key.toLowerCase() === 'x'` in place of `keycode.isEventKey(event, 'x')` etc.

May i ask is it possible fix at this stage?
